Rewrite FIXME to note that only tvOS and watchOS fails to compile some API tests

LumioseSil opened the pull request:
#### 9e90ceb499eb09b79c6525161cfc1901b6390212
<pre>
Rewrite FIXME to note that only tvOS and watchOS fails to compile some API tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=251928">https://bugs.webkit.org/show_bug.cgi?id=251928</a>

Reviewed by Jonathan Bedard.

We no longer compile for iOS 12 or macOS 10.14, but this ifdef also
meant that tvOS and watchOS did not have to run these tests, so I
reworded the FIXME to make that more clear.

*Source/JavaScriptCore/API/tests/testapi.mm:(testObjectiveCAPIMain):
Change FIXME and preprocessor directive

Canonical link: <a href="https://commits.webkit.org/262365@main">https://commits.webkit.org/262365@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9ced99addd2dbdd24815dcddae87838c72967451

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/688 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/707 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/735 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/905 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/606 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/756 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/792 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/845 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/695 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/661 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/645 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/864 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/731 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/640 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/672 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/605 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/619 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/661 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/680 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/651 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/644 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/711 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/604 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/646 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/149 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/333 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/661 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/712 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/143 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/JavaScriptCore/API/tests/testapi.mm:
```diff
@@ -649,9 +649,8 @@ static void testObjectiveCAPIMain()
-// Older platforms ifdef the type of some selectors so these tests don't work.
-// FIXME: Remove this when we stop building for macOS 10.14/iOS 12.
-#if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101500) || (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 130000)
+// FIXME: These tests fail on tvOS and watchOS
```
Is this something that you tested and confirmed? If not, it's better to not state this as a fact, but to say something like "Does this actually fail on watchOS and tvOS?" or "Need to confirm whether this fails on watchOS and tvOS."

I would expect that this condition was just wrongly written from the start, and then it became obvious after some mass replacement. Didn't check git history though.

AZero13 replied:
I tested and confirmed this.

AZero13 replied:
As in, I removed this check and ran all the tests and empirically determined this.

aproskuryakov replied:
Could you please tell us more? Did you run in simulator or on the device, and what was the failure?

AZero13 replied:
The failure was that the tests did not even build.

AZero13 replied:
error: cannot initialize a parameter of type 'JSValueProperty' (aka 'NSString *') with an lvalue of type 'JSValue *__strong'

stwrt reviewed (commented):
This looks good to me. I made a similar patch to do this, since I didn't realize this one existed. 

Need a reviewer to approve though.

JonWBedard reviewed (approved):
We no longer support macOS 10.14 and iOS 12, so we should simplify this.

LumioseSil committed:
Rewrite FIXME to note that only tvOS and watchOS fails to compile some API tests
https://bugs.webkit.org/show_bug.cgi?id=251928

Reviewed by Jonathan Bedard.

We no longer compile for iOS 12 or macOS 10.14, but this ifdef also
meant that tvOS and watchOS did not have to run these tests, so I
reworded the FIXME to make that more clear.

*Source/JavaScriptCore/API/tests/testapi.mm:(testObjectiveCAPIMain):
Change FIXME and preprocessor directive

Canonical link: https://commits.webkit.org/262365@main
